pkg/controller/managedresources/merger.go

```diff
@@ -69,14 +93,59 @@ func mergeDeployment(scheme *runtime.Scheme, oldObj, newObj runtime.Object) erro
 	}
 
 	// Do not overwrite a Deployment's '.spec.replicas' if the new Deployment's '.spec.replicas'
-	// field is unset.
-	if newDeployment.Spec.Replicas == nil {
+	// field is unset or the Deployment is scaled by either an HPA or HVPA.
+	if newDeployment.Spec.Replicas == nil || preserveReplicas {
 		newDeployment.Spec.Replicas = oldDeployment.Spec.Replicas
 	}
 
+	mergePodTemplate(&oldDeployment.Spec.Template, &newDeployment.Spec.Template, preserveResources)
+
 	return scheme.Convert(newDeployment, newObj, nil)
 }
 
+func mergePodTemplate(oldPod, newPod *corev1.PodTemplateSpec, preserveResources bool) {
+	// Do not overwrite a PodTemplate's resource requests / limits if it is scaled by an HVPA
```

Comment:
Can you already exit early here right at the beginning if `preserveResources` is `false`?
Then you don't have to pass it to `mergeContainer`, or?

Reply:
Sure, even better 👍